### PR TITLE
[5.5] Refactor resolve method to make more readable

### PR DESCRIPTION
**Pull request by georgehanson**
In the pull request I have altered the `resolve` method within the `Resource` class. It seems a bit odd in my opinion to be setting a variable to itself `$data = $data`. Therefore I have altered this slightly to remove it and make it more readable (in my opinion)

### CODE REVIEW

File: src/Illuminate/Http/Resources/Json/Resource.php [hunk 1]
```diff
@@ -91,12 +91,12 @@ public function resolve($request = null)
             $request = $request ?: Container::getInstance()->make('request')
         );
 
-        if (is_array($data)) {
-            $data = $data;
-        } elseif ($data instanceof Arrayable || $data instanceof Collection) {
-            $data = $data->toArray();
-        } elseif ($data instanceof JsonSerializable) {
-            $data = $data->jsonSerialize();
+        if ($data instanceof Arrayable || $data instanceof Collection) {
```
Comment (JosephSilber) [inline src/Illuminate/Http/Resources/Json/Resource.php]: A `Collection` is `Arrayable`.

File: src/Illuminate/Http/Resources/Json/Resource.php [hunk 2]
```diff
@@ -91,12 +91,12 @@ public function resolve($request = null)
...
+            return $this->filter((array) $data->toArray());
```
Comment (JosephSilber) [inline src/Illuminate/Http/Resources/Json/Resource.php]: Why are you casting it to an array again?

File: src/Illuminate/Http/Resources/Json/Resource.php (hunk 1 above)
Comment (GeorgeHanson) [reply src/Illuminate/Http/Resources/Json/Resource.php]: Fair point

File: src/Illuminate/Http/Resources/Json/Resource.php (hunk 2 above)
Comment (GeorgeHanson) [reply src/Illuminate/Http/Resources/Json/Resource.php]: Good spot. I only re-arranged the code, I didn't look too much into what it was doing. I will amend this

**Comment by ConnorVG** [comment]
I feel like:
```php
if ($data instanceof Arrayable) {
    $data = $data->toArray();
} elseif ($data instanceof JsonSerializable) {
    $data = $data->jsonSerialize();
}

return $this->filter((array) $data);
```

Better fits Laravel's code style.

File: src/Illuminate/Http/Resources/Json/Resource.php [hunk 3]
```diff
@@ -91,15 +91,13 @@ public function resolve($request = null)
...
 
-        if (is_array($data)) {
-            $data = $data;
-        } elseif ($data instanceof Arrayable || $data instanceof Collection) {
-            $data = $data->toArray();
+        if ($data instanceof Arrayable) {
+            $data = $this->filter($data->toArray());
         } elseif ($data instanceof JsonSerializable) {
-            $data = $data->jsonSerialize();
+            $data = $this->filter($data->jsonSerialize());
```
Comment (JosephSilber) [inline src/Illuminate/Http/Resources/Json/Resource.php]: Why are you calling `filter` here & above?

**Comment by ConnorVG** [comment]
You need to not call `filter` so many times (no early return or pre-filtering) and you need to account for arrayable instances that don't implement `Arrayable` or `JsonSerializable`, such as `stdClass`.

`stdClass` is neither `Arrayable` or `JsonSerializable` but can be cast to an array `(array) $obj` and can be serialized `json_encode($obj)`.

Keep the `(array)`.

**Comment by GeorgeHanson** [comment]
Changes have been made

File: src/Illuminate/Http/Resources/Json/Resource.php (hunk 3 above)
Comment (GeorgeHanson) [reply src/Illuminate/Http/Resources/Json/Resource.php]: Oops...

**Comment by ConnorVG** [comment]
Cool. Now this PR simply just removes excess code without losing functionality or changing the results.

😄

**Comment by GeorgeHanson** [comment]
Yup, got there in the end @ConnorVG 😆

**Comment by sisve** [comment]
Wouldn't it be appropriate to add some tests to ensure that the scenarios discussed here is supported in the future to? Anything that verifies that it still works with arrays, collections and stdObjects.

**Comment by taylorotwell** [comment]
Unless something is actually broken, I see no reason to change this. If there is a bug feel free to open another PR with a failing test and this change.

**Pull request closed without merging**